src/main/java/org/spongepowered/api/event/SpongeEventFactory.java

```diff
@@ -801,14 +804,16 @@ public static ProjectileLaunchEvent createProjectileLaunch(Game game, Cause caus
      * @param arguments The arguments provided
      * @param source The source of the command
      * @param command The command name
+     * @param result The result of the command
      * @return A new instance of the event
      */
-    public static CommandEvent createCommand(Game game, String arguments, CommandSource source, String command) {
+    public static CommandEvent createCommand(Game game, String arguments, CommandSource source, String command, Optional<CommandResult> result) {
```

Comment:
Don't use `Optional` in method parameters, instead, the parameter should be annotated with `@Nullable` and values.put should be `values.put("result", Optional.fromNullable(result));`